pkg/git/github_test.go

```diff
@@ -117,9 +125,11 @@ func TestGitHubWithNoAuthentication(t *testing.T) {
 
 // makeAPIServer is used during testing to create an HTTP server to return
 // fixtures if the request matches.
-func makeGitHubAPIServer(t *testing.T, authToken, wantPath, etag string, response []byte) *httptest.Server {
+func makeGitHubAPIServer(t *testing.T, authToken, wantPath, etag string, response []byte, tagResponse []byte) *httptest.Server {
```

Comment:
A couple of options here, you could make the responses into a map, with paths and responses which would simplify the lookup later in the function...

Maybe something like this:

```go
responses := map[string]response{
  "/path/to/tags": response{data: []byte{...}},
  "/path/to/branch": response{data: []byte{...}, etag: ...},
}
```
So, in the method, you'd try and find `r.URL.Path` in the map, if it doesn't exist, you 404, otherwise, you optionally write the ETag header if `response.etag != ""` and finally, write the data as the response body.